### PR TITLE
SDK-1407. Fix duplicates searching on outshares

### PR DESCRIPTION
**Pull request by jgandres**

### CODE REVIEW

**Review by Yenel — APPROVED** [review]
Tested, works fine.

File: src/megaapi_impl.cpp [hunk 1]
```diff
@@ -11774,9 +11774,15 @@ MegaNodeList* MegaApiImpl::search(MegaNode *n, const char* searchString, MegaCan
         if (target == MegaApi::SEARCH_TARGET_OUTSHARE)
         {
             // Search on outshares
+            std::set<MegaHandle>outsharesHandles;
```
Comment (sergiohs84) [inline src/megaapi_impl.cpp:11777]:
```suggestion
            std::set<MegaHandle> outsharesHandles;
```

File: src/megaapi_impl.cpp [hunk 2]
```diff
@@ -11774,9 +11774,15 @@ MegaNodeList* MegaApiImpl::search(MegaNode *n, const char* searchString, MegaCan
...
             unique_ptr<MegaShareList>shares (getOutShares(MegaApi::ORDER_NONE));
             for (int i = 0; i < shares->size() && !(cancelToken && cancelToken->isCancelled()); i++)
             {
+                if (outsharesHandles.find(shares->get(i)->getNodeHandle()) != outsharesHandles.end())
+                {
+                    continue;   // avoid duplicates
+                }
+                outsharesHandles.insert(shares->get(i)->getNodeHandle());
```
Comment (sergiohs84) [inline src/megaapi_impl.cpp:11785]: You can improve readability a bit:
```suggestion cpp
                handle h = shares->get(i)->getNodeHandle();
                if (outsharesHandles.find(h) != outsharesHandles.end())
                {
                    // shares list includes an item per outshare AND per sharee/user
                    continue;   // avoid duplicates
                }
                outsharesHandles.insert(h);
```
Also, should use the declared `h` in the next line too :)

**Review by sergiohs84 — CHANGES_REQUESTED** [review]
Good catch (and patch) ;)

**Pull request closed without merging**